cpp/src/table/table.cpp

```diff
@@ -33,10 +33,15 @@ table::table(table const& other) : _num_rows{other.num_rows()} {
 // Move the contents of a vector `unique_ptr<column>`
 table::table(std::vector<std::unique_ptr<column>>&& columns)
     : _columns{std::move(columns)} {
-  CUDF_EXPECTS(columns.size() > 0, "Invalid number of columns");
-  _num_rows = columns[0]->size();
-  for (auto const& c : _columns) {
-    CUDF_EXPECTS(c->size() == num_rows(), "Column size mismatch.");
+  if(num_columns() > 0) {
+    _num_rows = _columns[0]->size();
+    for (auto const& c : _columns) {
+      CUDF_EXPECTS(c, "Unexpected null column");
+      CUDF_EXPECTS(c->size() == num_rows(), "Column size mismatch.");
```

Comment:
I realized we'll need to change this to make my other suggestion work. 
```suggestion
      CUDF_EXPECTS(c->size() == _columns.first(), "Column size mismatch.");
```

Reply:
Ah, I missed it as well.